### PR TITLE
rc 6.0.0

### PR DESCRIPTION
**Pull request by apaparazzi0329**
This PR is the release candidate branch for the upcoming major release version 6.0.0

### CODE REVIEW

File: MIGRATION-V6.md
Comment (dpopp07) [inline MIGRATION-V6.md:20]: Didn't this happen in the last major release? We don't need to document it here if so

File: MIGRATION-V6.md
Comment (dpopp07) [inline MIGRATION-V6.md:43]: You might want to show use of the `then` pattern here, just for its similarity to the callback pattern. Then, you can also show the `async/await` pattern. For that case, I would remind the user that `await` must be used inside of an `async` function or just show the function definition in your example.

File: MIGRATION-V6.md
Comment (dpopp07) [inline MIGRATION-V6.md:48]: We're only dropping v10 so there's no need to document when v6 and v8 ended. Replace that with the end date of v10

File: MIGRATION-V6.md
Comment (dpopp07) [inline MIGRATION-V6.md:146]: I'm not going to look at all of these in detail. The format looks good, just make sure that you aren't including anything from the previous release on accident

File: assistant/v1.ts
Comment (dpopp07) [inline assistant/v1.ts:2]: Make sure you go through all of the copyrights before merging

File: visual-recognition/v3.ts
Comment (dpopp07) [inline visual-recognition/v3.ts:131]: @mediumTaj has this finally been fixed in the API definition or do we still need manual support for Array types here?

File: visual-recognition/v4.ts
Comment (dpopp07) [inline visual-recognition/v4.ts:122]: See my comment on the v3 SDK

File: speech-to-text/v1.ts
Comment (dpopp07) [inline speech-to-text/v1.ts:89]: This all looks good, there are just some style things. For example, there should be a space before the bracket on the try and the catch. A lot of statements are also missing semicolons.

```suggestion
        try {
```

File: speech-to-text/v1.ts
Comment (dpopp07) [inline speech-to-text/v1.ts:90]:
```suggestion
          const res = await self.listCorpora(params);
```

I'm only leaving a comment here as a reminder but a lot of statements need semicolons at the end to conform to our style guide

**Review by dpopp07 — CHANGES_REQUESTED** [review]
This is looking good. There are a few things to fix before merging, re-tag me once these are done and I'll give it another look over

File: visual-recognition/v3.ts
Comment (dpopp07) [reply visual-recognition/v3.ts:131]: Actually Mike just said that he updated the code in the service to accept either an array or a comma-separated string. In that case, we don't need this change anymore. Is that true, Taj?

File: assistant/v1.ts
Comment (apaparazzi0329) [reply assistant/v1.ts:2]: Good catch

**Review by dpopp07 — COMMENTED** [review]
Thanks for fixing that stuff up. I've left a number of comments on the tests that are not ultimately _required_. My thinking is that if you are going through the process of modernizing all of the tests, it might be worth cleaning up a lot of the sloppy testing practices that exist. I've left a bunch of comments on the Assistant v1 test showing what I mean. I'll leave it up to you to decide how much you want to address

File: visual-recognition/v3.ts
Comment (mediumTaj) [reply visual-recognition/v3.ts:131]: Nope, we still need this as comma separated strings. Mike updated in visrec v4 but they didn't think it was worth it to update v3.

File: visual-recognition/v4.ts
Comment (mediumTaj) [reply visual-recognition/v4.ts:122]: funny thing about v4 - it supports these as both comma separated strings and separate form parts. I think they are generated as separate form parts so no handwritten change needed anymore for this.

File: MIGRATION-V6.md
Comment (mediumTaj) [inline MIGRATION-V6.md:152]: training  status should be removed with a handwritten change. This should be readonly

File: personality-insights/v3.ts
Comment (mediumTaj) [inline personality-insights/v3.ts]: Did we run both lint and fix copyrights here? there should be two dates

File: visual-recognition/v3.ts
Comment (dpopp07) [reply visual-recognition/v3.ts:131]: Thanks for the updates Taj! So Angelo, you'll need to retain this code for the major release.

File: personality-insights/v3.ts
Comment (apaparazzi0329) [reply personality-insights/v3.ts]: I did run both but it appears the copyright linter missed this one. I'll fix it manually and make sure to check every file

File: MIGRATION-V6.md
Comment (apaparazzi0329) [reply MIGRATION-V6.md:152]: Thank you for catching that

File: speech-to-text/v1.ts
Comment (apaparazzi0329) [reply speech-to-text/v1.ts:66]: Nothing gets past your eagle eyes 😂

File: .travis.yml
Comment (dpopp07) [inline .travis.yml:9]: You may want to confirm this because I could be wrong - but by removing the "if" statement about it not being a pull request, it looks like the `auth.js` file will be created and that would mean the integration tests would be run. We do not want to run the integration tests on PR builds, only Push builds.

Whenever I updated the credentials, I would just drop the changes to the `.travis.yml` file - they are just formatting changes and are not needed. You can probably do the same here. The only thing I'm not sure about is this new key (`$encrypted_02d9f3c197ce_key`) - I'm not sure why that's there or how you need to included it

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍 

As I mentioned before, there are some issues with the async/await pattern in the tests. These are not urgent and they can be addressed post-release. I'll help fix those up soon

**Pull request merged**
